packages/vite/src/node/server/openBrowser.ts

```diff
@@ -96,9 +96,7 @@ async function startBrowserProcess(
       if (openedBrowser) {
         // Try our best to reuse existing tab with AppleScript
         await execAsync(
-          `osascript openChrome.applescript "${encodeURI(
-            url,
-          )}" "${openedBrowser}"`,
+          `osascript openChrome.applescript "${url}" "${openedBrowser}"`,
```

Comment:
It seems this `encodeURI` was added in https://github.com/facebook/create-react-app/pull/2076 to allow URLs containing `"`. I think we should do `url.replaceAll('"', '%22')` here.

Reply:
Thank you for your review! I have carefully checked all the code calling the `openBrowser` method. The URL obtained from `server.resolvedUrls?.local[0] ?? server.resolvedUrls?.network[0]` and the path retrieved from `new URL(options.open, url).href` both already encode the `"`. Is it still necessary to add `url.replaceAll('"', '%22')` to encode `"` in this method?

Looking forward to your feedback!

Reply:
Ah, then it should be ok. Thanks for checking!

Reply:
There's a spot where we don't use `new URL()`. Maybe we should add it while we're at it: https://github.com/vitejs/vite/blob/b80daa7c0970645dca569d572892648f66c6799c/packages/vite/src/node/shortcuts.ts#L142-L150

This part is for the preview server, which it's handling it correctly this way, but not for its shortcuts: https://github.com/vitejs/vite/blob/b80daa7c0970645dca569d572892648f66c6799c/packages/vite/src/node/preview.ts#L260-L268

Reply:
@bluwy Both handles `resolvedUrls` the same way (passes `openBrowser` without `new URL()`). Are you suggesting to do `url = new URL(url).href`?

Reply:
@bluwy Thank you for pointing that out!

I’ve checked the part of the code you mentioned, and the URL obtained from `server.resolvedUrls?.local[0] ?? server.resolvedUrls?.network[0]` is already URL-encoded. So, I think there is no need to use `new URL()` in this case.

Reply:
You're right I got it mixed up. I was suggesting adding:

```ts
const path = 
       typeof server.config.preview.open === 'string' ? new URL(server.config.preview.open, url).href : url 
```

It's a different bug now that I look at it. We can do in a separate PR